frame/fast-unstake/src/lib.rs

```diff
@@ -15,37 +15,102 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! A pallet that's designed to JUST do the following:
+//! > Made with *Substrate*, for *Polkadot*.
 //!
-//! If a nominator is not exposed in any `ErasStakers` (i.e. "has not actively backed any
-//! validators in the last `BondingDuration` days"), then they can register themselves in this
-//! pallet, unstake faster than having to wait an entire bonding duration.
+//! [![github]](https://github.com/paritytech/substrate/frame/fast-unstake) -
+//! [![polkadot]](https://polkadot.network)
 //!
-//! Appearing in the exposure of a validator means being exposed equal to that validator from the
-//! point of view of the staking system. This usually means earning rewards with the validator, and
-//! also being at the risk of slashing with the validator. This is equivalent to the "Active
-//! Nominator" role explained in the
-//! [February Staking Update](https://polkadot.network/blog/staking-update-february-2022/).
+//! [polkadot]: https://img.shields.io/badge/polkadot-E6007A?style=for-the-badge&logo=polkadot&logoColor=white
+//! [github]: https://img.shields.io/badge/github-8da0cb?style=for-the-badge&labelColor=555555&logo=github
 //!
-//! This pallet works off the basis of `on_idle`, meaning that it provides no guarantee about when
-//! it will succeed, if at all. Moreover, the queue implementation is unordered. In case of
-//! congestion, no FIFO ordering is provided.
+//! # Fast Unstake Pallet
+//!
+//! A pallet to allow participants of the staking system (represented by [`Config::Staking`], being
+//! [`StakingInterface`]) to unstake quicker, if and only if they meet the condition of not being
+//! exposed to any slashes.
+//!
+//! ## Overview
+//!
+//! This pallet that's designed to JUST do the following:
```

Comment:
```suggestion
//! This pallet is designed to JUST do the following:
```

Although TBH I'm not sure what purpose this line serves, can it be removed?

Also, since we are improving the docs: I feel like the text that follows could use some structure, e.g. subsections or bullet points. Right now it seems (to me) like an unapproachable wall of text.

Reply:
I find your comment about this sentence fair, but not the rest. It is organized into a series of paragraphs and is following a logical chain. Perhaps if you describe further which part of it is unapprochable, I can improve. 

Nonetheless though, this is in realm of subjective measures harder to align on. I don't think we can ever make _everyone_ write docs that are understandable to _everyone_. Simply things like different ways of thinking and language levels will make that impossible. 

What we want to ensure is at least following the structure mentioned in #14115, and doing your best.

Reply:
Fair enough, it was indeed a subjective impression I got as someone new to this pallet. Reading my post again it sounds more critical than I intended it to be, sorry. 🙈 To be concrete, I would suggest adding subsections like e.g. "Registration" and "Checking". Feel free to ignore if you think that's unnecessary. The new docs are already awesome as is!